reporters/GlobalSummaryReporter.js

```diff
@@ -116,32 +115,40 @@ class GlobalSummaryReporter {
     // Pricefeed stats.
     this.priceEstimate = this.referencePriceFeed.getCurrentPrice();
 
-    // Initialize Uniswap pair address to get trade data for. Default pair token is the collateral token.
-    this.uniswapPairToken = this.uniswapPairOverride[this.toChecksumAddress(this.empContract.options.address)]
-      ? this.uniswapPairOverride[this.toChecksumAddress(this.empContract.options.address)]
+    // Initialize exchange pair address to get trade data for. Default pair token is the collateral token.
+    this.exchangePairToken = this.exchangePairOverride[this.toChecksumAddress(this.empContract.options.address)]
+      ? this.exchangePairOverride[this.toChecksumAddress(this.empContract.options.address)]
       : this.collateralContract;
-    this.uniswapPairDetails = await getUniswapPairDetails(
-      this.web3,
-      this.syntheticContract.address,
-      this.uniswapPairToken.address
-    );
-    this.uniswapPairAddress = this.uniswapPairDetails.pairAddress.toLowerCase();
-
-    // Set up Uniswap subgraph client and query latest update stats.
-    // If the `latestSwapBlockNumber` < `endBlockNumberForPeriod`, then set `endBlockNumberForPeriod = latestSwapBlockNumber`
-    // otherwise the graphQL client will throw an error when trying to access a block that it has not indexed yet.
-    // Its ok to use the latest swap's block number as the end-period block number because we are only querying swap data for this pair.
-    // If we were using `endBlockNumberForPeriod` to query data for some other pair as well,
-    // then we wouldn't be able to set `endBlockNumberForPeriod = latestSwapBlockNumber`
-    this.uniswapClient = getUniswapClient();
-    const latestSwap = (await this.uniswapClient.request(queries.LAST_TRADE_FOR_PAIR(this.uniswapPairAddress))).pairs[0]
-      .swaps[0].transaction;
-    this.latestSwapTimestamp = latestSwap.timestamp;
-    this.latestSwapBlockNumber = Number(latestSwap.blockNumber);
-    // Note: `endBlockNumberForPeriod` is the highest block number that we will manually query for.
-    if (this.endBlockNumberForPeriod > this.latestSwapBlockNumber) {
-      this.endBlockNumberForPeriod = this.latestSwapBlockNumber;
+
+    // Uniswap data:
+    if (this.uniswapPriceFeed) {
```

Comment:
Just curious -- if we were to add a balancer price feed, how would we differentiate between this being a uniswap feed and a balancer feed? I'm assuming we would need this information so we know what subgraph data to grab, etc, right?

Definitely not a problem that needs to be solved now -- just curious.

Reply:
I was thinking we could just add a `balancerPriceFeed` parameter in addition to the `uniswapPriceFeed`. This way the reporter will be able to support multiple exchanges in the future